PandorasBox/Features/Actions/AutoCordial.cs

```diff
@@ -0,0 +1,135 @@
+using PandorasBox.Features;
+using PandorasBox.FeaturesSetup;
+using ECommons.DalamudServices;
+using FFXIVClientStructs.FFXIV.Client.Game;
+using FFXIVClientStructs.FFXIV.Client.Game.UI;
+using Dalamud.Game;
+using Dalamud.Game.ClientState.Conditions;
+using System.Linq;
+using ECommons.Throttlers;
+using ImGuiNET;
+using System.Collections.Generic;
+
+namespace PandorasBox.Features.Actions
+{
+    public unsafe class AutoCordial : Feature
+    {
+        public override string Name => "Auto-Cordial";
+
+        public override string Description => "Automatically use a cordial when below the threshold.";
+
+        public override FeatureType FeatureType => FeatureType.Actions;
+
+        public Configs Config { get; private set; }
+
+        public override bool UseAutoConfig => false;
+
+        public class Configs : FeatureConfig
+        {
+            [FeatureConfigOption("GP Threshold", IntMin = 0, IntMax = 1000, EditorSize = 300)]
+            public int DefaultThreshold = 1;
+            public bool DirectionAbove = true;
+            public bool DirectionBelow = false;
+
+            [FeatureConfigOption("Invert Priority (Watered -> Regular -> Hi)")]
+            public bool InvertPriority = false;
+
+            [FeatureConfigOption("Prevent Overcap")]
+            public bool PreventOvercap = true;
+
+            [FeatureConfigOption("Use on Fisher")]
+            public bool UseOnFisher = false;
+        }
+
+        protected override DrawConfigDelegate DrawConfigTree => (ref bool _) =>
+        {
+            ImGui.PushItemWidth(300);
+            ImGui.SliderInt("GP Threshold", ref Config.DefaultThreshold, 1, 1000);
+            if (ImGui.RadioButton("Above", Config.DirectionAbove))
+            {
+                Config.DirectionAbove = true;
+                Config.DirectionBelow = false;
+            }
+            ImGui.SameLine();
+            if (ImGui.RadioButton("Below", Config.DirectionBelow))
+            {
+                Config.DirectionAbove = false;
+                Config.DirectionBelow = true;
+            }
+            ImGui.Checkbox("Invert Priority (Watered -> Regular -> Hi)", ref Config.InvertPriority);
+            ImGui.Checkbox("Prevent Overcap", ref Config.PreventOvercap);
+            ImGui.Checkbox("Use on Fisher", ref Config.UseOnFisher);
+        };
+
+        private bool WillOvercap(int gp_recovery)
+        {
+            return Svc.ClientState.LocalPlayer.CurrentGp + gp_recovery < Svc.ClientState.LocalPlayer.MaxGp;
+        }
```

Comment:
Consider looking to pass the item ID and check the sheets for a recovery amount.

Reply:
So I added a (commented out) draft of this in f74e13c just so you can see why I think this wouldn't work. The issue with this is that I don't believe recovery is present in the sheets, at least not as a regular value, just in the description. Also, would have to have more complicated handling in there for if the item is HQ, so I think the static values are probably the safer bet.

Reply:
Something like this? Not really sure how the Data{HQ} column should be handled.
`internal static (string Name, uint Id, uint ItemActionID, uint NQGP, uint HQGP)[] cordials;`
```cordials = Svc.Data.GetExcelSheet<Item>()
      .Where(row => row.Name.ToString().Contains("cordial")
      .Select(row => (
          Name: row.Name,
          Id: row.RowId,
          ItemActionID: row.ItemAction,
          NQGP: Svc.Data.GetExcelSheet<ItemAction>()
              .First(actionRow => actionRow.RowId == row.ItemActionID).Data[0],
          HQGP: Svc.Data.GetExcelSheet<ItemAction>()
              .First(actionRow => actionRow.RowId == row.ItemActionID).Data{HQ}[0]
      )).ToArray());```